### DISCUSSION TITLE
Android App connection issue

### DISCUSSION THREAD

**Discussion by 74tommyboy** [advplyr/audiobookshelf#1028] [Q&A]
Hello everyone,

I am having issues accessing the app outside my network.

Here is my setup:
- ABS running on docker in Synology NAS. I have my own domain and am going through cloud flare for DNS. I am also running tailscale on the Nas.
- In cloudflare, I have created an A record with my NAS' TS IP address.
- I have also created a CNAME record pointing to the ABS sub domain (audiobooks.mynas.com).

On my NAS, I've created a reverse proxy entry with a websocket.
   - Source is the subdomain:443 
   - Destination is: local IP:13378 I've enabled HSTS

While on the network, I can access the web app as well as the android app.

If I ping the subdomain, I get the TS IP address of my NAS, so I know that I'm getting to the NAS.

I also tried creating a page redirect on CF, but that didn't make a difference.

I am able to connect from an external network if I use the https://audiobooks.mynas.com:port

when I enter https://audiobooks.mynas.com:port into the android app, I get the error message that server could not be pinged.

Any help would be greatly appreciated.

**Comment by digiSal** [comment]
Did you figure this out?

**Comment by PassMeTheMustard** [comment]
Why don't you just use the tailscale name (or IP) and port 13378 when connecting to the app. Obviously you will need to have tailscale on your phone or device and have it enabled.
I've been using this since I installed ABS (since only a couple of weeks) and it works seamlessly so far at home or out and about. Haven't got around to other remote options yet but with tailscale I haven't really felt the need.

**Comment by shaftspanner** [comment]
Did you ever figure this out?  I've just moved from SWAG directly accessing my server via my domain to a Cloudflare tunnel accessing ABS via a new domain.  I can access the web app without any problems via the cloudflare tunnel, but the android app won't connect.

In the app, I type in the URL, hit submit, get a brief spinner, then the screen returns to enter the URL - no error message

On the server, the logs show the following:
```2023-04-18 11:25:42 DEBUG [Server] User Offline mike
2023-04-18 11:25:42  INFO  [Server] Socket MlpkmU6a2RmbnB1yAADX disconnected from client "mike" after 51426ms (Reason: transport close)
2023-04-18 11:25:44  INFO  [Server] Socket Connected JuXSLFoZYW6vfyEcAADZ
2023-04-18 11:25:44  DEBUG  [Server] User Online mike
2023-04-18 11:25:44  DEBUG  [DB] Updated user: 1
```
After this sequence, the android app definitely isn't connected.  I'm assuming the Socket Connected message at 11:25:44 is my web client on the PC.

Edited for clarity
